backend/layers/processing/dataset_metadata_update.py

```diff
@@ -0,0 +1,344 @@
+            h5ad_job = Process(
+                target=DatasetMetadataUpdater.update_h5ad,
+                args=(
+                    self.artifact_bucket,
+                    self.datasets_bucket,
+                    artifact_uris[DatasetArtifactType.H5AD],
+                    current_dataset_version,
+                    new_artifact_key_prefix,
+                    new_dataset_version_id,
+                    metadata_update,
+                ),
+            )
+            artifact_jobs.append(h5ad_job)
+            h5ad_job.start()
+        else:
+            self.logger.error(f"Cannot find labeled H5AD artifact uri for {current_dataset_version_id}.")
+            self.update_processing_status(new_dataset_version_id, DatasetStatusKey.H5AD, DatasetConversionStatus.FAILED)
+
+        if DatasetArtifactType.RDS in artifact_uris:
+            self.logger.info("Main: Starting thread for rds update")
+            rds_job = Process(
+                target=DatasetMetadataUpdater.update_rds,
+                args=(
+                    self.artifact_bucket,
+                    self.datasets_bucket,
+                    artifact_uris[DatasetArtifactType.RDS],
+                    new_artifact_key_prefix,
+                    new_dataset_version_id,
+                    metadata_update,
+                ),
+            )
+            artifact_jobs.append(rds_job)
+            rds_job.start()
+        elif current_dataset_version.status.rds_status == DatasetConversionStatus.SKIPPED:
+            self.update_processing_status(new_dataset_version_id, DatasetStatusKey.RDS, DatasetConversionStatus.SKIPPED)
+        else:
+            self.logger.error(
+                f"Cannot find RDS artifact uri for {current_dataset_version_id}, and Conversion Status is not SKIPPED."
+            )
+            self.update_processing_status(new_dataset_version_id, DatasetStatusKey.RDS, DatasetConversionStatus.FAILED)
+
+        if DatasetArtifactType.CXG in artifact_uris:
+            self.logger.info("Main: Starting thread for cxg update")
+            cxg_job = Process(
+                target=DatasetMetadataUpdater.update_cxg,
+                args=(
+                    self.artifact_bucket,
+                    self.datasets_bucket,
+                    artifact_uris[DatasetArtifactType.CXG],
+                    f"s3://{self.cellxgene_bucket}/{new_artifact_key_prefix}.cxg",
+                    new_dataset_version_id,
+                    metadata_update,
+                ),
+            )
+            artifact_jobs.append(cxg_job)
+            cxg_job.start()
+        else:
+            self.logger.error(f"Cannot find cxg artifact uri for {current_dataset_version_id}.")
+            self.update_processing_status(new_dataset_version_id, DatasetStatusKey.CXG, DatasetConversionStatus.FAILED)
+
+        # blocking call on async functions before checking for valid artifact statuses
+        [j.join() for j in artifact_jobs]
```

Comment:
Actually, I'm wrong. After reading the logic more closely, it is evident that the ConversionStatus is being set at the end of every subprocess worker job, and this is then interpreted by the main thread which will appropriately update the processing status as a failure. We do not need additional checks.

But, since this is a step function and we'll want to be able to quickly find the failed runs in the console, we should probably throw an error [here](https://github.com/chanzuckerberg/single-cell-data-portal/blob/88a8eec8926ea32af5d333a8ab0b41165bec6a88/backend/layers/processing/dataset_metadata_update.py#L311-L313).